gen/src/namespace_organizer.rs

```diff
@@ -0,0 +1,40 @@
+use crate::syntax::Api;
+use proc_macro2::Ident;
+use std::collections::BTreeMap;
+
+pub(crate) struct NamespaceEntries<'a> {
+    pub(crate) entries: Vec<&'a Api>,
+    pub(crate) children: BTreeMap<&'a Ident, NamespaceEntries<'a>>,
+}
+
+pub(crate) fn sort_by_namespace(apis: &[Api]) -> NamespaceEntries {
+    let api_refs = apis.iter().collect::<Vec<_>>();
```

Comment:
Because I'm trying to convert from `&[Api]` to `&[&Api]` in effect. All the code inside these functions operates purely on references to the `Api`s but that's not what we're given - we've given a slice of the actual APIs.

I feel certain there's a better way to do this but I haven't spotted it. Let me know if you know!